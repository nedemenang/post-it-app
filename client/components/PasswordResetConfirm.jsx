import React, { Component } from 'react';
import QueryString from 'query-string';
import '../public/style.scss';
import { receiveErrors, confirmPasswordReset } from '../actions/AppActions';


/**
 * Redirects to home page after resetting password
 *
 * @return {void} returns void
 */
function redirectToHome() {
  setTimeout(() => {
    window.location.href = '/#';
  }, 2000); // will call the function after 2 secs.
}

/**
 * @class PasswordResetConfirm
 *
 * @extends {Component}
 */
class PasswordResetConfirm extends Component {

/**
 * Calls the confirm password reset function
 *
 * @return {void} return void
 *
 * @param {object} event event object
 *
 * @memberof PasswordResetConfirm
 */
  reset(event) {
  //
    event.preventDefault();
    const password = this.state.password.trim();
    const confirmPassword = this.state.confirmPassword.trim();

    if (password === '') {
      receiveErrors('Please insert password');
    } else if (confirmPassword === '') {
      receiveErrors('Please insert confirmation password');
    } else if (confirmPassword !== password) {
      receiveErrors('Password and confirmation passwords do not match');
    } else {
      const parsed = QueryString.parse(location.search);
      const resetObject = {
        code: parsed.oobCode,
        newPassword: password
      };
      confirmPasswordReset(resetObject);
      redirectToHome();
    }
  }


  /**
   * Function to handle change in password reset field
   *
   * @param {object} event object
   *
   * @memberof PasswordResetConfirm
   */
  handlePasswordChange(event) {
    this.setState({ password: event.target.value });
  }


  /**
   *Function to handle change in password confirm field
   *
   * @param {object} event object
   *
   * @memberof PasswordResetConfirm
   */
  handleResetPasswordConfirmChange(event) {
    this.setState({ confirmPassword: event.target.value });
  }

  /**
   * Creates an instance of PasswordResetConfirm.
   * @param {object} props props object
   * @memberof PasswordResetConfirm
   */
  constructor(props) {
    super(props);
    this.state = {
      password: '',
      confirmPassword: ''
    };

    this.reset = this.reset.bind(this);
  }

  /**
   *
   * Renders password reset confirm page
   * @returns {JSX} returns password reset confirm page
   * @memberof PasswordResetConfirm
   */
  render() {
    return (
      <div>
  <div className="login-page">
      <div className="form">
    <form className="login-form">
    <h3>Reset Password</h3>
      <input type="password"
      id="password"
      onChange ={this.handlePasswordChange.bind(this)}
      placeholder="password"/>
      <input type="password"
      id="confirmPassword"
      onChange = {this.handleResetPasswordConfirmChange.bind(this)}
      placeholder="Confirm password"/>
      <button className="button" onClick={this.reset}>Save</button>
            <br/>
       <br/>
    </form>

  </div>
</div>
</div>
    );
  }

}

export default PasswordResetConfirm;
