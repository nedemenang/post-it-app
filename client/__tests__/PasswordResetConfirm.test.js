import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PasswordResetConfirm from '../components/PasswordResetConfirm.jsx';

describe('the password reset component', () => {
  let mountedComponent;

  const PasswordResetConfirmItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<PasswordResetConfirm />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = PasswordResetConfirmItem();
    expect(component).toBeDefined();
  });

  it('should match snapshot test', () => {
    const component = PasswordResetConfirmItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it('should update state on password field change', () => {
    const input = PasswordResetConfirmItem().find('#password').first();
    input.simulate('change', { target: {
      value: 'password1'
    } });
    expect(PasswordResetConfirmItem().state().password).toBe('password1');
  });

  it('should update state on confirm password field change', () => {
    const input = PasswordResetConfirmItem().find('#confirmPassword').first();
    input.simulate('change', { target: {
      value: 'password1'
    } });
    expect(PasswordResetConfirmItem().state()
    .confirmPassword).toBe('password1');
  });

  it('should call function on submit of form', () => {
    const component = shallow(<PasswordResetConfirm />);
    const preventDefault = jest.fn();
    component.find('button').simulate('click', { preventDefault });
    expect(preventDefault).toBeCalled();
  });
});
