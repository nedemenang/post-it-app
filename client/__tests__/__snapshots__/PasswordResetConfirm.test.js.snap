// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the password reset component should match snapshot test 1`] = `
<div>
  <div
    className="login-page"
  >
    <div
      className="form"
    >
      <form
        className="login-form"
      >
        <h3>
          Reset Password
        </h3>
        <input
          id="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
        />
        <input
          id="confirmPassword"
          onChange={[Function]}
          placeholder="Confirm password"
          type="password"
        />
        <button
          className="button"
          onClick={[Function]}
        >
          Save
        </button>
        <br />
        <br />
      </form>
    </div>
  </div>
</div>
`;
